File: problems/0020_Valid_Parentheses/Bruce0020.py
# 第二十题

# Solution one in python

# 下述内容有问题。需要更改

class Solution:
    def isValid(self, s):
	    stack = []
		dict = {"]":"[","{":"}",")":"("}
		for char in s:
		    if char in dict.values():
			    stack.append(char)
			elif char in dict.keys():
			    if stack == [] or dict[char] != stack.pop():
				    return False
			else:
			    return False
		return stack == []


# Solution Two in java


		
		

File: problems/0020_Valid_Parentheses/Bruce0020.java
/******
这是用java写堆栈

*******/


class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<Character>();
			//这里的for(char c:chars)就是定义一个遍历字符c，让它分别等于字符串数组chars里面的各个字符，然后执行下面的语句，当c被赋值为chars里面所有字符各一次后，就会退出这个循环.
        for (char c : s.toCharArray()){
            if (c == '(')
                stack.push(')');
            else if (c == '{')
                stack.push('}');
            else if (c == '[')
                stack.push(']');
            else if (stack.isEmpty() || stack.pop()!= c)
                return false;
        }
        return stack.isEmpty();}
}
	


